flutter/lib/src/localizations.dart

```diff
@@ -0,0 +1,13 @@
+import 'package:flutter/cupertino.dart';
+import 'package:meta/meta.dart';
+
+GlobalKey<NavigatorState> sentryNavigatorKey = GlobalKey<NavigatorState>();
+
+@internal
+Locale? retrieveWidgetLocale() {
+  final BuildContext? context = sentryNavigatorKey.currentContext;
+  if (context != null) {
+    return Localizations.localeOf(context);
+  }
+  return null;
+}
```

Comment:
Doesn't it make more sense that this is a property of `SentryFlutterOptions` and user are able to assign it during setup? That way the users are also more flexible, since they can assign their own navigator key.

E.g. something like:
```dart
await SentryFlutter.init((options) {
  options.navigatorKey = userDefinedNavigatorKey;
});
```

The `retrieveWidgetLocale` should also use [`maybeOf`](https://api.flutter.dev/flutter/widgets/Localizations/maybeLocaleOf.html), in order to not throw in case there's no localization set.
```dart
Locale? retrieveWidgetLocale() {
  final BuildContext? context = sentryNavigatorKey.currentContext;
  if (context != null) {
    return Localizations.maybeLocaleOf(context);
  }
  return null;
}
```

Reply:
Yeah, should be better with a user defined key, thx for the catch